src/wp-admin/site-editor.php

```diff
@@ -34,6 +35,43 @@
 	}
 }
 
+// Validate postId and postType.
+if ( isset( $_GET['postId'] ) && $post_type_param ) {
+	switch ( $post_type_param ) {
+		case 'page':
+			$post = get_post( (int) $_GET['postId'] );
+
+			if ( null === $post || 'page' !== get_post_type( $post ) ) {
+				wp_die( __( 'Invalid page ID.' ) );
+			}
+			break;
+
+		case 'wp_block':
+			$post = get_post( (int) $_GET['postId'] );
+
+			if ( null === $post || 'wp_block' !== get_post_type( $post ) ) {
+				wp_die( __( 'Invalid pattern ID.' ) );
+			}
+			break;
+
+		case 'wp_template':
+			$block_template = get_block_template( $_GET['postId'] );
+
+			if ( null === $block_template ) {
+				wp_die( __( 'Invalid template ID.' ) );
+			}
+			break;
+
+		case 'wp_template_part':
+			$block_template = get_block_template( $_GET['postId'], 'wp_template_part' );
+
+			if ( null === $block_template ) {
+				wp_die( __( 'Invalid template part ID.' ) );
```

Comment:
[Same here](https://github.com/WordPress/wordpress-develop/pull/7188/files#r1792315558), but maybe should link to the Templates page.